core/object/class_db.h

```diff
@@ -263,7 +268,7 @@ class ClassDB {
 	static bool is_parent_class(const StringName &p_class, const StringName &p_inherits);
 	static bool can_instantiate(const StringName &p_class);
 	static bool is_virtual(const StringName &p_class);
-	static Object *instantiate(const StringName &p_class);
+	static Object *instantiate(const StringName &p_class, bool p_require_real_class = false);
```

Comment:
I would try to leave the instantiate function clean, and instead add another one like `instantiate_no_extension` or something, so the code is easier to understand.

Reply:
Ok! I've make it so there's `ClassDB::instantiate(const StringName &p_class)` and `ClassDB::instantiate_no_placeholders(const StringName &p_class)` in my latest push